mass_mailing_custom_unsubscribe/controllers/main.py

```diff
@@ -43,6 +44,7 @@ def mailing(self, mailing_id, email=None, res_id=None, token="", **post):
             "Called `mailing()` with: %r",
             (mailing_id, email, res_id, token, post))
         reasons = request.env["mail.unsubscription.reason"].search([])
+        res_id = res_id and int(res_id)
```

Comment:
:thinking: Humm this line makes me think... Are we actually supporting the case of `res_id=None` or `res_id=""`? If not, then I guess this should just be:

```suggestion
        res_id = int(res_id)
```

This way, we fail fast in case there's some evil `res_id` incoming.

Reply:
It's not failing if False/None is passed, so this should be this way. It's copied from upstream anyway.